pkg/segment/query/processor/streamstatscommand.go

```diff
@@ -18,26 +18,942 @@
 package processor
 
 import (
+	"container/list"
+	"fmt"
+	"io"
+	"math"
+	"sort"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/cespare/xxhash"
+	"github.com/siglens/siglens/pkg/common/dtypeutils"
 	"github.com/siglens/siglens/pkg/segment/query/iqr"
 	"github.com/siglens/siglens/pkg/segment/structs"
+	"github.com/siglens/siglens/pkg/segment/utils"
+	segutils "github.com/siglens/siglens/pkg/segment/utils"
+	putils "github.com/siglens/siglens/pkg/utils"
 )
 
 type streamstatsProcessor struct {
-	options *structs.StreamStatsOptions
+	options          *structs.StreamStatsOptions
+	currentIndex     int
+	currentBucketKey string
 }
 
 func (p *streamstatsProcessor) Process(iqr *iqr.IQR) (*iqr.IQR, error) {
-	panic("not implemented")
+	if iqr == nil {
+		return nil, io.EOF
+	}
+
+	log.Infof("streamstats: starting processing for %d records with window=%d",
+		iqr.NumberOfRecords(), p.options.Window)
+
+	outIQR := iqr.BlankCopy()
+
+	measureAggs := p.options.MeasureOperations
+	if p.options.GroupByRequest != nil {
+		measureAggs = p.options.GroupByRequest.MeasureOperations
+	}
+
+	if p.options.RunningStreamStats == nil {
+		p.options.RunningStreamStats = make(map[int]map[string]*structs.RunningStreamStatsResults)
+	}
+
+	allColumns, err := iqr.ReadAllColumns()
```

Comment:
Why do we need to read all columns ? The idea behind this design is to use the inputIQR and add the extra fields created accordingly using `AppendKnownValues`.